File: code/community/1301690985805254662/managing-short-lived-api-keys.js
require('dotenv').config();
const deepgram = require('@deepgram/sdk');

// Retrieve API key from environment variables
const apiKey = process.env.DEEPGRAM_API_KEY;

// Initialize Deepgram SDK client
const deepgramClient = new deepgram.Deepgram(apiKey);

// Function to simulate API key deletion logic
async function deleteApiKey(keyId) {
    console.log(`Deleting API key with id: ${keyId}`);
    // Insert actual API key deletion logic here using your own backend or management service
}

// Example usage
(async () => {
    const keyId = process.env.DEEPGRAM_KEY_ID;
    // Assume the API key is used here
    
    // Delete the API key immediately after use
    await deleteApiKey(keyId);
})();

// Ensure to have DEEPGRAM_API_KEY and DEEPGRAM_KEY_ID in your `.env` file.

File: code/community/1301690985805254662/managing-short-lived-api-keys.go
package main

import (
	"fmt"
	"os"
)

func main() {
	// Read the Deepgram API Key from an environment variable
	apiKey := os.Getenv("DEEPGRAM_API_KEY")

	// Create a new Deepgram client (Placeholder as actual function depends on SDK)
	client := NewDeepgramClient(apiKey)

	// Example to delete an API key after use
	keyID := os.Getenv("DEEPGRAM_KEY_ID")
	
	// Assume the key is used prior in logic
	deleteAPIKey(client, keyID)
}

// NewDeepgramClient is a placeholder to represent Deepgram client creation
func NewDeepgramClient(apiKey string) *DeepgramClient {
	// Initialize and return a new Deepgram client
	return &DeepgramClient{apiKey: apiKey}
}

// DeepgramClient is a struct representing a Deepgram client
// This is a placeholder for the existing struct within Deepgram SDK
type DeepgramClient struct {
	apiKey string
}

// Function to delete an API key
func deleteAPIKey(client *DeepgramClient, keyID string) {
	fmt.Printf("Deleting API key with ID: %s\n", keyID)
	// Implement actual key deletion logic here if available
}

// Assume this logic calls a backend service or directly uses the Deepgram SDK
// to manage the lifecycle of API keys.